### PR TITLE
Feature/#907 kick ban context menu

### PR DESCRIPTION
**Pull request by Geosearchef**

### CODE REVIEW

File: src/main/java/com/faforever/client/api/dto/LobbyGroup.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.faforever.client.api.dto;
+
+import com.github.jasminb.jsonapi.annotations.Id;
+import com.github.jasminb.jsonapi.annotations.Type;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+@Getter
+@Deprecated
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/api/dto/LobbyGroup.java]: Mark depecated once the new system is there not before that....

File: src/main/java/com/faforever/client/moderator/ModeratorService.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  private final FafService fafService;
+  private final FafApiAccessor fafApiAccessor;
+
+  public ModeratorService(FafService fafService, FafApiAccessor fafApiAccessor) {
+    this.fafService = fafService;
+    this.fafApiAccessor = fafApiAccessor;
+  }
+
+
+  public CompletableFuture<Boolean> isModerator() {
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/moderator/ModeratorService.java:24]: cache this... this is called from every chat user item i dont wanan have server crashing if somebody scolls user list

File: src/main/java/com/faforever/client/moderator/ModeratorService.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+public class ModeratorService {
+  private static CompletableFuture isModerator;
+  private final FafService fafService;
+  private final FafApiAccessor fafApiAccessor;
+
+  public ModeratorService(FafService fafService, FafApiAccessor fafApiAccessor) {
+    this.fafService = fafService;
+    this.fafApiAccessor = fafApiAccessor;
+  }
+
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/moderator/ModeratorService.java:23]: remove new line

File: src/main/java/com/faforever/client/moderator/BanDialogController.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+public class BanDialogController implements Controller<Node> {
+
+  private final ModeratorService moderatorService;
+  private final I18n i18n;
+  public Pane banDialogRoot;
+  public Button dismissButton;
+  public Label userLabel;
+  public TextField reasonText;
+  public TextField durationText;
+  public ChoiceBox periodTypeChoiceBox;
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/moderator/BanDialogController.java]: supports generics use those (at least i think so)

File: src/main/java/com/faforever/client/chat/ChatUserContextMenuController.java [hunk 5]
```diff
@@ -309,4 +344,17 @@ public void onRemoveCustomColor() {
...
+  public void onKickGame() {
+    moderatorService.closePlayersGame(getPlayer().getId());
+  }
+
+  public void onKickLobby() {
+    moderatorService.closePlayersLobby(getPlayer().getId());
+  }
+
+
+  public void consumer(ActionEvent actionEvent) {
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/chat/ChatUserContextMenuController.java:346]: What does this do? Java doc or rename maybe

File: src/main/java/com/faforever/client/moderator/BanDialogController.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  public Button banButton;
+  public Label successsLabel;
+  private JFXDialogLayout dialogLayout;
+  private ObjectProperty<Player> player;
+
+  private Runnable closeListener;
+
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/moderator/BanDialogController.java:44]: remove new line

File: src/main/resources/theme/chat/chat_user_context_menu.fxml [hunk 7]
```diff
@@ -48,16 +48,23 @@
...
+      <MenuItem fx:id="banItem" mnemonicParsing="false" onAction="#onBan" visible="false" text="%chat.userContext.ban"/>
+
+      <CustomMenuItem fx:id="avatarPickerMenuItem" hideOnClick="false" mnemonicParsing="false">
       <content>
       </content>
       <content>
           <JFXComboBox fx:id="avatarComboBox" maxWidth="1.7976931348623157E308"/>
       </content>
     </CustomMenuItem>
+
```
Comment (1-alex98) [inline src/main/resources/theme/chat/chat_user_context_menu.fxml]: emove new lines

File: src/main/java/com/faforever/client/chat/ChatUserContextMenuController.java [hunk 8]
```diff
@@ -117,6 +127,23 @@ public void initialize() {
     colorPicker.focusedProperty().addListener((observable, oldValue, newValue) -> chatUserContextMenuRoot.setAutoHide(!newValue));
   }
 
+  private void initModerator() {
+    moderatorService.isModerator().thenAccept(isModerator -> {
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/chat/ChatUserContextMenuController.java]: lets unit test both cases and whether kickGameItem and co are set to correct visibility

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/moderator/ModeratorService.java (hunk 2 above)
Comment (Geosearchef) [reply src/main/java/com/faforever/client/moderator/ModeratorService.java:24]: why?
The completable future is initialized with the first call and then the call to the API will never ever trigger again.

Comment (1-alex98) [reply src/main/java/com/faforever/client/moderator/ModeratorService.java:24]: true did not see that

File: src/main/java/com/faforever/client/chat/ChatUserContextMenuController.java (hunk 5 above)
Comment (Geosearchef) [reply src/main/java/com/faforever/client/chat/ChatUserContextMenuController.java:346]: no, it consumes an action event, exactly what is says

Comment (1-alex98) [reply src/main/java/com/faforever/client/chat/ChatUserContextMenuController.java:346]: whats the effect?

**Comment by 1-alex98** [comment]
fix travis

**Comment by 1-alex98** [comment]
Added the missing tests fo u

**Pull request merged**